File: app/src/main/java/com/phucnguyen/lovereminder/ui/dialog/ChangeThemeDialog.java
package com.phucnguyen.lovereminder.ui.dialog;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatDialogFragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.phucnguyen.lovereminder.R;
import com.phucnguyen.lovereminder.ui.adapter.ThemesAdapter;

public class ChangeThemeDialog extends AppCompatDialogFragment {
    public interface ThemeDialogListener {
        void onThemeDialogChanged(int themeId);
    }
    private RecyclerView rcvThemes;
    private ThemesAdapter adapter;

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());

        LayoutInflater inflater = getActivity().getLayoutInflater();
        View view = inflater.inflate(R.layout.dialog_change_theme, null);
        connectView(view);

        builder.setView(view)
                .setTitle("Thay đổi theme")
                .setNegativeButton("Hủy", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {

                    }
                })
                .setPositiveButton("Xác nhận", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        dismiss();
                        int themeId = adapter.getCurrentSelectedThemeId();
                        ((ThemeDialogListener) getTargetFragment()).onThemeDialogChanged(themeId);
                    }
                });

        return builder.create();
    }

    private void connectView(View view) {
        rcvThemes = view.findViewById(R.id.rcvThemes);

        rcvThemes.setLayoutManager(new LinearLayoutManager(view.getContext(), RecyclerView.HORIZONTAL, false));
        adapter = new ThemesAdapter(view.getContext());
        rcvThemes.setAdapter(adapter);
    }
}

File: app/src/main/java/com/phucnguyen/lovereminder/ui/fragment/dialog/ChangeThemeDialog.kt
package com.phucnguyen.lovereminder.ui.fragment.dialog

import android.app.Dialog
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatDialogFragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.phucnguyen.lovereminder.R
import com.phucnguyen.lovereminder.ui.adapter.ThemesAdapter

class ChangeThemeDialog : AppCompatDialogFragment() {
    interface ThemeDialogListener {
        fun onThemeDialogChanged(themeId: Int)
    }

    private lateinit var rcvThemes: RecyclerView
    private lateinit var adapter: ThemesAdapter
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(
            activity!!
        )
        val inflater = activity!!.layoutInflater
        val view = inflater.inflate(R.layout.dialog_change_theme, null)
        connectView(view)
        builder.setView(view)
            .setTitle("Thay đổi theme")
            .setNegativeButton("Hủy") { dialogInterface, i -> }
            .setPositiveButton("Xác nhận") { dialogInterface, i ->
                dismiss()
                val themeId = adapter.currentSelectedThemeId
                (targetFragment as ThemeDialogListener?)!!.onThemeDialogChanged(themeId)
            }
        return builder.create()
    }

    private fun connectView(view: View) {
        rcvThemes = view.findViewById(R.id.rcvThemes)
        rcvThemes.layoutManager = LinearLayoutManager(
            view.context,
            RecyclerView.HORIZONTAL,
            false
        )
        adapter = ThemesAdapter(view.context)
        rcvThemes.adapter = adapter
    }
}